lib/gooddata/models/segment.rb

```diff
@@ -61,7 +62,7 @@ def all(opts = {})
         client = domain.client
 
         results = client.get(domain.segments_uri + '/segments')
-        GoodData::Helpers.get_path(results, %w(segments items)).map { |i| client.create(GoodData::Segment, i.merge('domain' => domain)) }
+        GoodData::Helpers.get_path(results, %w(segments items)).map { |i| client.create(GoodData::Segment, i, domain: domain) }
```

Comment:
Refers to 'client' more than self (maybe move it to another class?) (FeatureEnvy)